### DISCUSSION TITLE
Sign in user on server based on user_id to bypass RLS

### DISCUSSION THREAD

**Discussion by pczern** [supabase/supabase#19920] [Questions] [answered]
Hello Community,
I want to allow users to create API Keys in my application with which they can access my implemented API.
Therefore I want users to only read/write rows they have access to via RLS. I have seen the auth.uid() function and I have a table linking user_id's to api keys.

For now I have only found a way to login with email and password, not via user_id.

But I need to retrieve a **jwt to log in for the supabase client library?**
How can I retrieve this **jwt on the server side from a user_id**?
Are there alternative ways to authenticate the user via user_id besides jwt?

<img width="1086" alt="image" src="https://github.com/supabase/supabase/assets/10871302/c85498f7-75c8-4ed7-ad49-74671c042103">

**Comment by GaryAustin1** [comment, marked as answer]
You probably will need to just mint a jwt with any jwt library using the Supabase jwt secret.
There are a few required claims: https://supabase.com/docs/learn/auth-deep-dive/auth-deep-dive-jwts#jwts-in-supabase
Then you use the minted jwt in the Authorization header as 'bearer jwt' which is a createClient option.
This works fine for general API stuff but it doesn't play with normal auth sessions and clients like auth-helpers/ssr.

**Reply by pczern** [reply]
thank you, it works perfectly
